superdesk/core/app.py

```diff
@@ -105,15 +111,55 @@ def stop(self):
 
         self.mongo.stop()
         self._imported_modules.clear()
+        self._remove_app()
         self._running = False
 
+    def _store_app(self):
+        from flask import current_app
+
+        try:
+            setattr(current_app, "async_app", self)
+        except RuntimeError:
+            # Flask context not available
+            pass
+
+        global _global_app
+        _global_app = self
+
+    def _remove_app(self):
+        from flask import current_app
+
+        try:
+            setattr(current_app, "async_app", None)
+        except RuntimeError:
+            # Flask context not available
+            pass
+
+        global _global_app
+        _global_app = None
```

Comment:
What is the reason for the global here? Just curious. Normally what I've read in the past is that the usages of globals are not encouraged.

Reply:
The `_global_app` is stored in the module's root scope (as it has to be stored somewhere, for later retrieval).
Without the use of `global`, assignment to that variable won't work inside a function scope.
I agree I don't like it much, and ATM is mainly there for tests.

We can improve this later (I also have ideas I'd like to implement, regarding module loading, circular references, module file structure, app passing etc).